### PR TITLE
migrate formatDistance to TS

### PR DESCRIPTION
**Pull request by a-korzun**
Fixes https://github.com/date-fns/date-fns-typescriptify/issues/32

### CODE REVIEW

**Review by tan75 — CHANGES_REQUESTED** [review]
Please resolve conflicts

**Comment by tan75** [comment]
Hello @a-korzun,

could you please resolve conflicts in your code?

File: src/formatDistance/test.ts [hunk 1]
```diff
@@ -1,7 +1,6 @@
-// @flow
 /* eslint-env mocha */
 
-import assert from 'power-assert'
```
Comment (tan75) [inline src/formatDistance/test.ts:4]: Is there a reason for a this switch from power-assert to assert?

**Review by tan75 — CHANGES_REQUESTED** [review]

File: src/formatDistance/test.ts (hunk 1 above)
Comment (a-korzun) [reply src/formatDistance/test.ts:4]: Nope. Updated

**Comment by a-korzun** [comment]
> Hello @a-korzun,
> 
> could you please resolve conflicts in your code?

Hi! Don't see any conflicts, may you provide a bit more information about it?
Just updated PR in order to fix migration from `power-assert`.

**Comment by tan75** [comment]
> > Hello @a-korzun,
> > could you please resolve conflicts in your code?
> 
> Hi! Don't see any conflicts, may you provide a bit more information about it?
> Just updated PR in order to fix migration from `power-assert`.

@a-korzun 
It turns out we are using assert now rather than power-assert and you were right when changing it. 
Could you please switch it back? 
Apologies for the confusion!

**Comment by a-korzun** [comment]
@tan75 Ok, done.

**Review by tan75 — APPROVED** [review]
Thank you for your contribution!

File: src/formatDistance/index.ts [hunk 2]
```diff
@@ -107,41 +107,48 @@ var MINUTES_IN_TWO_MONTHS = 86400
...
  */
-export default function formatDistance(dirtyDate, dirtyBaseDate, dirtyOptions) {
+
+export interface Options {
+  includeSeconds?: boolean
+  addSuffix?: boolean
+  locale?: Locale
+}
+
+export default function formatDistance(dirtyDate: Date | number, dirtyBaseDate: Date | number, dirtyOptions?: Options) {
```
Comment (kossnocorp) [inline src/formatDistance/index.ts:130]: Please import and use shared types from https://github.com/date-fns/date-fns/blob/master/src/types.ts

```suggestion
export default function formatDistance(dirtyDate: Date | number, dirtyBaseDate: Date | number, options : LocaleOptions & { includeSeconds?: boolean, addSuffix?: boolean } = {}) {
```

I appreciate the effort of exporting the type, but unless it's a common approach in the whole library, we don't want to do it. So please remove the exported type and put the type inline as I suggested. In the future, we might opt to export option types, but we'll need to think about the naming convention first.

**Review by kossnocorp — CHANGES_REQUESTED** [review]
Looks good, thanks! Please address my comment about the options type.

**Comment by a-korzun** [comment]
No problem! Done

**Review by tan75 — APPROVED** [review]
Thank you!

**Pull request merged**